src/components/counter.jsx

```diff
@@ -15,33 +15,48 @@ class Counter extends Component {
   // }
   
   render() {
-    const {counter} = this.props;
+    const {counter, counters} = this.props;
     return (
       <div>
         {this.props.children}
         <span className={this.getBadgeClasses()}>{this.formatCount()}</span>
         <button
           id={this.getBtnIncrId(counter.id)}
-          onClick={() => this.props.onIncrement(this.props.counter)}
+          onClick={this.incrementCounter.bind(this)}
+          disabled={this.isBtnIncrDis(counter, counters)}
           className="btn btn-secondary btn-sm"
         >Increment
         </button>
         <button 
           id={this.getBtnDecrId(counter.id)}
-          onClick={() => this.props.onDecrement(this.props.counter)}
+          onClick={this.decrementCounter.bind(this)}
+          disabled={this.isBtnDecrDis(counter, counters)}
           className="btn btn-secondary btn-sm m-2"
         >Decrement
         </button>
         <button 
           id={this.getBtnDelId(counter.id)}
-          onClick={() => this.props.onDelete(this.props.counter.id)} 
+          onClick={this.deleteCounter.bind(this)}
```

Comment:
Technically, calling `.bind(this)` returns a new function\*, so this has the same issue of telling React there's a new prop value for `onClick` on every re-render. I give a tip below on how to avoid needing to use `.bind` altogether, but otherwise the (old) best practice for JS classes in general is to do something like this from the constructor:
```
this.decrementCounter = this.decrementCounter.bind(this);
```

\* try this in the console drawer
```
const func = (function(){});
func.bind(this) === func.bind(this)
```